tests/http/server.cpp

```diff
@@ -505,3 +522,109 @@ BOOST_AUTO_TEST_CASE(urlcasesensitivity)
+{
+    zeroeq::http::Server server;
+    Foo foo;
+    server.handle( foo );
+    server.handlePUT( "bla/bar", zeroeq::PUTFunc( [] { return true; } ));
+
+    bool running = true;
+    std::thread thread( [&]() { while( running ) server.receive( 100 ); });
+
+    Client client( server.getURI( ));
+    client.test( "GET /registry HTTP/1.0\r\n\r\n",
+                 std::string( "HTTP/1.0 200 OK\r\n" +
+                              cors_headers +
+                              "Content-Length: 63\r\n\r\n" ) +
+                              "{\n   \"bla/bar\" : [ \"PUT\" ],\n" +
+                              "   \"test/foo\" : [ \"GET\", \"PUT\" ]\n}\n",
+                 __LINE__ );
+
+    running = false;
+    thread.join();
+}
+
+BOOST_AUTO_TEST_CASE(object_schema)
+{
+    zeroeq::http::Server server;
+    Foo foo;
+    server.handleGET( foo );
+
+    bool running = true;
+    std::thread thread( [&]() { while( running ) server.receive( 100 ); });
+
+    Client client( server.getURI( ));
+    client.test( "GET /test/foo/schema HTTP/1.0\r\n\r\n",
+                 std::string( "HTTP/1.0 200 OK\r\n" +
+                              cors_headers +
+                              "Content-Length: 26\r\n\r\n" ) +
+                              "{\n  '_notified' : 'bool'\n}",
+                 __LINE__ );
+
+    running = false;
+    thread.join();
+}
+
+BOOST_AUTO_TEST_CASE(event_schema)
+{
+    zeroeq::http::Server server;
+    const std::string schema = "{ \"value\" : \"boolean\" }";
+    server.handleGET( "bla/bar", schema,
+        zeroeq::GETFunc( [] { return std::string( "{ \"value\" : true }"); } ));
+    server.handlePUT( "bla/foo", schema,
+                      zeroeq::PUTFunc( []() { return true; } ));
+
+    bool running = true;
+    std::thread thread( [&]() { while( running ) server.receive( 100 ); });
+
+    Client client( server.getURI( ));
+    client.test( "GET /bla/bar/schema HTTP/1.0\r\n\r\n",
+                 std::string( "HTTP/1.0 200 OK\r\n" +
+                              cors_headers +
+                              "Content-Length: 23\r\n\r\n" ) +
+                              schema,
+                 __LINE__ );
+    client.test( "GET /bla/foo/schema HTTP/1.0\r\n\r\n",
+                 std::string( "HTTP/1.0 200 OK\r\n" +
+                              cors_headers +
+                              "Content-Length: 23\r\n\r\n" ) +
+                              schema,
+                 __LINE__ );
+
+    running = false;
+    thread.join();
+}
+
+BOOST_AUTO_TEST_CASE(event_no_schema)
+{
+    zeroeq::http::Server server;
+    server.handleGET( "bla/bar",
+        zeroeq::GETFunc( [] { return std::string( "{ \"value\" : true }"); } ));
+
+    bool running = true;
+    std::thread thread( [&]() { while( running ) server.receive( 100 ); });
+
+    Client client( server.getURI( ));
+    client.test( "GET /bla/bar/schema HTTP/1.0\r\n\r\n", error404, __LINE__ );
+
+    running = false;
+    thread.join();
+}
```

Comment:
all GETs mentioned should return meaningful values, i.e., should not fail. I don't see in the spec that using an object named schema is forbidden, and I don't see a reason why it should be. idem for registry, except in the global namespace.